anndata/core/anndata.py

```diff
@@ -296,17 +297,20 @@ def __init__(
         if X is not None:
             self._X = X
             self._var = var
-            self._varm = varm
+            self._varm = AxisArrays(self, 1, varm)
         else:
             self._X = None if adata.isbacked else adata.X.copy()
             self._var = adata.var.copy()
-            self._varm = adata.varm.copy()
+            self._varm = AxisArrays(self, 1, adata.varm.copy())
 
     @property
     def X(self):
         if self._adata.isbacked:
             if not self._adata.file.isopen: self._adata.file.open()
-            X = self._adata.file['raw.X']
+            if "raw/X" in self._adata.file._file.h5f:
```

Comment:
Please implement `AnnDataFileManager.__contains__` instead, then this could be:

```py
x_key = 'raw/X' if 'raw/X' in self._adata.file else 'raw.X'
X = self._adata.file[x_key]
```

or you could also implement `AnnDataFileManager.get` and do

```py
X = self._adata.file.get('raw/X') or self._adata.file['raw.X']
```

Reply:
I've implemented `__contains__` but I've kept the multiline conditionals since I've added error messages if neither is present.